plugins/defaultinvitationplugin/defaultinvitationplugin.cpp

```diff
@@ -318,11 +318,13 @@ QString DefaultInvitationPlugin::pluginName() const
 
 QString DefaultInvitationPlugin::icon() const
 {
+    d->mErrorCode = ServiceInterface::ErrorOk;
```

Comment:
This error api is strange. The ServiceInterface defines it as "Sometimes the true/false is not enough, so in case of false more details can be obtained".

Could interpret that as only  (some?) bool methods should be setting the error code? Or how would one know when to expect something.

Reply:
Let's rather limit the scope of the errors reported by the error code rather than expand it. I don't see anything useful coming out of icon getter failing and the code calling handling that depending on the error code.

What could make sense is these "True if OK, false otherwise" methods giving more info in case of failure. One could, though, question whether they should just an error code instead of boolean then. If I'd try to justify current way, I could say this allows code like "if (!operation())  { /* handle error*/ }

Reply:
Ok, I see. I've thus let it as it is now.